mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -1796,6 +1797,10 @@ void ModuleImport::processFunctionAttributes(llvm::Function *func,
       attr.isStringAttribute())
     funcOp.setTargetCpuAttr(StringAttr::get(context, attr.getValueAsString()));
 
+  if (llvm::Attribute attr = func->getFnAttribute("tune-cpu");
+      attr.isStringAttribute())
+    funcOp.setTuneCpuAttr(StringAttr::get(context, attr.getValueAsString()));
```

Comment:
Could you also add a small test that verifies the import / export works independent from Flang?

The target_cpu attributes are tested here:
```
llvm-project/mlir/test/Target/LLVMIR/target-cpu.mlir
llvm-project/mlir/test/Target/LLVMIR/Import/target-cpu.ll
```
It may make sense to add the tests in the same files and maybe rename them to target-and-tune-cpu.mlir/ll or have a separate set of files if the attributes are unrelated (which is not my impression).

Thanks!

Reply:
Thank you so much for this suggestion and for the pointers for how to do this kind of test.  I was struggling to figure out the proper mechanism and you helped greatly.  I have added two separate tests for tune-cpu instead of combining with the target-cpu ones because I think it's cleaner, but I am happy to combine if you prefer. Just let me know.  Thanks again for your help!

Reply:
Nice!

Separate tests are fine.